New Test  (304167@main): [ Tahoe Release x86_64 ] TestWebKitAPI.LockdownMode.CanaryFontSucceedsInFontParser is a flaky failure

vitorroriz opened the pull request:
#### b9d9f8123bd8fe32825ee2d11449917d0fd9ec93
<pre>
New Test  (304167@main): [ Tahoe Release x86_64 ] TestWebKitAPI.LockdownMode.CanaryFontSucceedsInFontParser is a flaky failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=304014">https://bugs.webkit.org/show_bug.cgi?id=304014</a>
<a href="https://rdar.apple.com/166319528">rdar://166319528</a>

Reviewed by Wenson Hsieh.

Not waiting for the font to load might cause flakiness.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/LockdownModeFonts.mm:
(TestWebKitAPI::loadFontFace):
(TestWebKitAPI::waitForFontLoad):
(TestWebKitAPI::TEST(LockdownMode, NotAllowedFontLoadingAPI)):
(TestWebKitAPI::TEST(LockdownMode, AllowedFontLoadingAPI)):
(TestWebKitAPI::TEST(LockdownMode, NotSupportedFontLoadingAPI)):
(TestWebKitAPI::TEST(LockdownMode, DISABLED_ImmediateParsedViaSafeFontParser)):
(TestWebKitAPI::TEST(LockdownMode, DISABLED_CanaryFontSucceedsInFontParser)):

Canonical link: <a href="https://commits.webkit.org/305115@main">https://commits.webkit.org/305115@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca736ddb8e7f2f5aa9662e0dbb5b3ebc1e441323

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137542 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9903 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48829 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/145299 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90515 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6c875f03-77a6-4c0c-a43f-a4fa361c2f02) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/139414 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10606 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/10033 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/105191 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76810 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3ec85837-99f4-44bb-bb23-7c74b7da66f5) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140487 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7871 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/123269 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/86044 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7507 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/5230 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5882 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116877 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41425 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/148063 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9585 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41979 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113570 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9603 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/8082 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113914 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28913 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7427 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119508 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/64241 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9634 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37561 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9365 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9574 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9426 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/LockdownModeFonts.mm:
```diff
@@ -85,10 +85,22 @@
             "let font = new FontFace('WebFont', fontData);"
-            "document.fonts.add(font);"
-            "target.style.setProperty('font-family', 'WebFont, Helvetica');", encoded.get()]];
+            "void font.load().then(() => {"
```
Is it possible to de-duplicate some of this code? The logic for setting `fontLoadComplete` asynchronously is really similar in (at least) three of these instances (in the error case for one of them, we set font-family on the target element's inline style but in the other we don't):

```
void font.load().then(() => {
  document.fonts.add(font);
  target.style.setProperty('font-family', 'WebFont, Helvetica');
  window.fontLoadComplete = true;
}).catch(() => {
  target.style.setProperty('font-family', 'WebFont, Helvetica');
  window.fontLoadComplete = true;
});

---

void font.load().then(() => {
  document.fonts.add(font);
  target.style.setProperty('font-family', 'WebFont, Helvetica');
  window.fontLoadComplete = true;
}).catch(() => {
  window.fontLoadComplete = true;
});
```

vitorroriz committed:
New Test  (304167@main): [ Tahoe Release x86_64 ] TestWebKitAPI.LockdownMode.CanaryFontSucceedsInFontParser is a flaky failure
https://bugs.webkit.org/show_bug.cgi?id=304014
rdar://166319528

Reviewed by Wenson Hsieh.

Not waiting for the font to load might cause flakiness.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/LockdownModeFonts.mm:
(TestWebKitAPI::loadFontFace):
(TestWebKitAPI::waitForFontLoad):
(TestWebKitAPI::TEST(LockdownMode, NotAllowedFontLoadingAPI)):
(TestWebKitAPI::TEST(LockdownMode, AllowedFontLoadingAPI)):
(TestWebKitAPI::TEST(LockdownMode, NotSupportedFontLoadingAPI)):
(TestWebKitAPI::TEST(LockdownMode, DISABLED_ImmediateParsedViaSafeFontParser)):
(TestWebKitAPI::TEST(LockdownMode, DISABLED_CanaryFontSucceedsInFontParser)):

Canonical link: https://commits.webkit.org/305115@main
